homeassistant/components/camera/verisure.py

```diff
@@ -70,42 +64,40 @@ def check_imagelist(self):
-                    hub.my_pages.smartcam.download_image(self._device_id,
-                                                         self._new_image_id,
-                                                         self._file_path)
-                    self._smartcam_old_dict = self._smartcam_dict
-                    if self._image_id:
-                        _LOGGER.debug('self._image_id=%s', self._image_id)
-                        self._delete_image = os.path.join(self._file_path,
-                                                          '{}{}'.format(
-                                                              self._image_id,
-                                                              '.jpg'))
-                        _LOGGER.debug('Deleting %s', self._delete_image)
-                        os.remove(self._delete_image)
-                        self._image_id = self._new_image_id
-                        self._image = os.path.join(self._file_path,
-                                                   '{}{}'.format(
-                                                       self._image_id,
-                                                       '.jpg'))
-                    else:
-                        self._image = os.path.join(self._file_path,
-                                                   '{}{}'.format(
-                                                       self._new_image_id,
-                                                       '.jpg'))
-                        self._image_id = self._new_image_id
+        for self._device_id in self._smartcam_dict:
+            self._images = list(self._smartcam_dict.items())
+            self._new_image_id = self._images[0][1][0]
+            _LOGGER.debug('self._device_id=%s, self._images=%s, '
+                          'self._new_image_id=%s', self._device_id,
+                          self._images, self._new_image_id)
+            if (self._new_image_id == '-1' or
+                self._image_id == self._new_image_id):
+                _LOGGER.debug('The image is the same, or loading image_id')
+                return
+            _LOGGER.debug('Download new image %s', self._new_image_id)
+            hub.my_pages.smartcam.download_image(self._device_id,
+                                                 self._new_image_id,
+                                                 self._directory_path)
+            self._smartcam_old_dict = self._smartcam_dict
+            if self._image_id:
+                _LOGGER.debug('self._image_id=%s', self._image_id)
+                self._delete_image = os.path.join(self._directory_path,
+                                                 '{}{}'.format(
+                                                      self._image_id,
+                                                      '.jpg'))
+                _LOGGER.debug('Deleting %s', self._delete_image)
+                os.remove(self._delete_image)
+                self._image_id = self._new_image_id
```

Comment:
```
self._image = os.path.join(self._directory_path,
                                           '{}{}'.format(
                                              self._new_image_id,
                                              '.jpg'))
self._image_id = self._new_image_id
```
Is the same in both if and else and will always execute. It can be placed after the if instead.